crates/api/src/v1/monitor.rs

```diff
@@ -0,0 +1,188 @@
+//! Types relating to the monitor API.
+
+use crate::Status;
+use serde::{de::Unexpected, Deserialize, Serialize, Serializer};
+use std::borrow::Cow;
+use thiserror::Error;
+use warg_crypto::hash::AnyHash;
+use warg_crypto::signing;
+use warg_protocol::registry::RegistryLen;
+
+/// Represents checkpoint verification response in one of the following states:
+#[derive(Serialize, Deserialize)]
+#[serde(tag = "state", rename_all = "camelCase")]
+#[allow(clippy::large_enum_variant)]
+pub enum CheckpointVerificationResponse {
+    /// The checkpoint is verified.
+    #[serde(rename_all = "camelCase")]
+    Verified,
+    /// The checkpoint is unverified but should retry verification.
+    #[serde(rename_all = "camelCase")]
+    Retry {
+        /// Instructs to retry but wait the specified number of seconds.
+        wait_seconds: u16,
+    },
+    /// The checkpoint is unverified and should not retry.
+    #[serde(rename_all = "camelCase")]
+    Unverified,
+}
+
+/// Represents a monitor API error.
+#[non_exhaustive]
+#[derive(Debug, Error)]
+pub enum MonitorError {
+    /// The provided checkpoint log length is greater than the most recent checkpoint. The monitor
+    /// is expected to only return this after it is certain that it has the most recent checkpoint.
+    #[error("checkpoint log length `{0} is greater than the most recent checkpoint")]
+    CheckpointNotFound(RegistryLen),
+    /// The checkpoint signature keyId was not found.
+    #[error("checkpoint signature keyId `{0}` is invalid for checkpoint")]
+    CheckpointSignatureKeyIdInvalid(signing::KeyID),
+    /// The checkpoint signature invalid.
+    #[error("checkpoint signature `{0}` is invalid")]
+    CheckpointSignatureInvalid(signing::Signature),
```

Comment:
> If a registry doesn't feel like verifying that an old key had permission at the correct time to sign this checkpoint it can simply not verify and return Unverified.

After thinking through... If a key is revoked, it could have been compromised. We should return an error indicating that `KeyID` was revoked the permission to sign checkpoints.

Reply:
Revised